clients/drcachesim/common/trace_entry.h

```diff
@@ -747,6 +747,123 @@ struct schedule_entry_t {
+    SYSCALL_PT_ENTRY_TYPE_SYSCALL_ARGS_NUM,
+    SYSCALL_PT_ENTRY_TYPE_MAX = 6
+} syscall_pt_entry_type_t;
+
+START_PACKED_STRUCTURE
+struct _syscall_pt_entry_t {
+    union {
+        struct {
+            uint64_t pid : 61;
+            uint64_t type : 3;
+        } pid;
+        struct {
+            uint64_t tid : 61;
+            uint64_t type : 3;
+        } tid;
+        union {
+            struct {
+                uint64_t data_size : 61;
+                uint64_t type : 3;
+            } pt_metadata_boundary;
+            struct {
+                uint64_t data_size : 61;
+                uint64_t type : 3;
+            } pt_data_boundary;
+        };
+        struct {
+            uint64_t sysnum : 61;
+            uint64_t type : 3;
+        } sysnum;
+        struct {
+            uint64_t id : 61;
+            uint64_t type : 3;
+        } syscall_id;
+        struct {
+            uint64_t args_num : 61;
+            uint64_t type : 3;
+        } syscall_args_num;
+        uint64_t combined_value;
+    };
+} END_PACKED_STRUCTURE;
+typedef struct _syscall_pt_entry_t syscall_pt_entry_t;
+
+/**
+ * The per-thread metadata and all syscalls' PT data are stored in one PT DATA
+ * Buffer(PDB). The PDB format is:
+ * +----------+----------------------+
+ * |PDB Header|PT_METADATA or PT_DATA|
+ * +----------+----------------------+
+ *
+ * The PDB Header is a list of syscall_pt_entry_t. There are three types of PDB Header:
+ * a. The format of per-thread metadata's PDB header is:
+ * +---+---+--------------------+
+ * |pid|tid|pt_metadata_boundary|
+ * +---+---+--------------------+
+ *
+ * b. The format of syscalls' PT data's PDB header is:
+ * +---+---+----------------+------+----------+---------------------+---------+
+ * |pid|tid|pt_data_boundary|sysnum|syscall_id|syscall_args_boundary|arguments|
+ * +---+---+----------------+------+----------+---------------------+---------+
+ */
+
+/* The header of per-thread metadata buffer contains 3 syscall_pt_entry_t instances:
+ * 1. The first instance is used to store the pid.
+ * 2. The second instance is used to store the tid.
+ * 3. The 3rd instance is used to store the output buffer's type.
+ */
+#    define PT_METADATA_PDB_HEADER_ENTRY_NUM 3
+#    define PT_METADATA_PDB_HEADER_SIZE \
+        (PT_METADATA_PDB_HEADER_ENTRY_NUM * sizeof(syscall_pt_entry_t))
+#    define PT_METADATA_PDB_DATA_OFFSET PT_METADATA_PDB_HEADER_SIZE
+
+/* The header of each syscall's PT data buffer contains 12 syscall_pt_entry_t
+ * instances:
+ * 1. The first instance is used to store the pid.
+ * 2. The second instance is used to store the tid.
+ * 3. The 3rd instance is used to store the output buffer's type.
+ * 4. The 5th instance is used to store the sysnum.
+ * 5. The 5th instance is employed for preserving the sequence of syscalls.
+ * 6. The 6th instance is used to store the number of all input arguments.
+ * 7-12. The remaining instances are used to store the size of each input argument.
```

Comment:
Since syscall_args_num stored the count of args, do we need to add all six even in cases where the syscall had fewer args?

Reply:
Done.